plugins/kubernetes/app/models/concerns/kubernetes/deploy_yaml.rb

```diff
@@ -30,10 +34,43 @@ def raw_template
       @raw_template ||= build.file_from_repo(kubernetes_role.config_file)
     end
 
-    def add_labels
-      labels.each do |key, value|
+    # This key replaces the default kubernetes key: 'deployment.kubernetes.io/podTemplateHash'
+    # This label is used by kubernetes to identify a RC and corresponding Pods
+    def set_rc_unique_label_key
+      template.spec.uniqueLabelKey = CUSTOM_UNIQUE_LABEL_KEY
```

Comment:
This doesn't seem very unique to me :) If there are multiple deployments going on, won't they interfere with each others replication controllers?